packages/artifact/src/internal/download/download-artifact.ts

```diff
@@ -89,8 +99,51 @@ export async function streamExtractExternal(
         clearTimeout(timer)
         reject(error)
       })
-      .pipe(unzip.Extract({path: directory}))
-      .on('close', () => {
+      .pipe(unzip.Parse())
+      .pipe(
+        new stream.Transform({
+          objectMode: true,
+          transform: async (entry, _, callback) => {
+            const fullPath = path.normalize(path.join(directory, entry.path))
```

Comment:
`const fullPath = path.resolve(directory, entry.path)`

will `path.resolve()` be an better option, assume `entry.path` should be a relative path?

Reply:
That makes sense to me! I think `download-artifact` ensures that the path is absolute, but it doesn't look like toolkit does any validation.

Reply:
Ah, doing that throws off the check below for the directory path 😞

Reply:
Would you mind if I changed it back? On second thought, it might be better to keep this generic and not assume path for anyone utilizing this package. For download-artifact, we have the absolute path: https://github.com/actions/download-artifact/blob/348754975ef0295bfa2c111cba996120cfdf8a5d/src/download-artifact.ts#L38

Reply:
i thought `directory` is absolute path, looks like it can be relative...

Reply:
Directory is absolute when it comes from `download-artifact`, but isn't guaranteed absolute from other sources.